Core/Business/Context.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Business;
+
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+class Context extends ParameterBag
+{
+    /**
+     * @var Context
+     */
+    private static $instance = null;
+
+    /**
+     * @var \Core_Foundation_IoC_Container
+     */
+    private static $container = null;
+
+    /**
+     * Get current instance of filled context (singleton)
+     *
+     * @param \Core_Foundation_IoC_Container $container The service container
+     * @return Context
+     */
+    final public static function getInstance(\Core_Foundation_IoC_Container &$container = null)
```

Comment:
Hum, maybe...
I think you should always need to access Context in your actions (language, logged user, country, etc...), not always container (that is just a Service provider). So I do it in this way...

But you can even access the container from the Controller (linked as a ref through static attribute inside the Controller parent class), no need to get it through the Context.
I needed the container to access services outside of the controller scope when I have only Context available.

On the other hand, Context::getInstance() is here and easy to use, but the container is not a singleton that can be retrieve by a "Container::getInstance()" for now. Maybe I have to add this singleton support. What do you think about this?